File: 1662-check-if-two-string-arrays-are-equivalent/1662-check-if-two-string-arrays-are-equivalent.ts
function arrayStringsAreEqual(word1: string[], word2: string[]): boolean {
    let s1 = "";
    let s2 = "";
    
    let i = 0;
    let j = 0;
    while (i < word1.length || j < word2.length) {
        if (i < word1.length) {
            s1 += word1[i];
            i++;
        }
        if (j < word2.length) {
            s2 += word2[j];
            j++;
        }
    }
    
    return s1 === s2;
};

File: 1662-check-if-two-string-arrays-are-equivalent/1662-check-if-two-string-arrays-are-equivalent.go
func arrayStringsAreEqual(word1 []string, word2 []string) bool {
	w1 := ""
	w2 := ""

	i := 0
	j := 0

	for i < len(word1) || j < len(word2) {
		if i < len(word1) {
			element := word1[i]
			w1 += element
			i++
		}
		if j < len(word2) {
			element := word2[j]
			w2 += element
			j++
		}
	}

	return w1 == w2
}
